src/model_signing/__init__.py

```diff
@@ -14,19 +14,119 @@
+  and file shards. The result of hashing is a manifest, a listing of hashes for
+  every object in the model.
+- `model_signing.signing`: responsible with taking the manifest and generating a
+  signature, based on a signing configuration. The signing configuration can
+  select the method used to sign as well as the parameters.
+- `model_signing.verifying`: responsible with taking a signature and verifying
+  it. If the cryptographic parts of the signature can be validated, the
+  verification layer would return an expanded manifest which can then be
+  compared agains a manifest obtained from hashing the existing model. If the
+  two manifest don't match then the model integrity was compromised and the
+  `model_signing` package detected that.
+
+The first two of these components allows configurability but can also be used
+directly, with a default configuration. The only difference is for the
+verification component where we need to configure the verification method since
+there are no sensible defaults that can be used.
+
+Signing can be done using the default configuration:
+
+```python
+model_signing.signing.sign("finbert", "finbert.sig")
+```
+
+This example generates the signature using Sigstore.
+
+Alternatively, a custom configuration can be selected, for both signing and
+hashing:
+
+```python
+model_signing.signing.Config().use_elliptic_key_signer(
+    private_key="key"
+).set_hashing_config(
+    model_signing.hashing.Config().set_ignored_paths(
+        paths=["README.md"], ignore_git_paths=True
+    )
+).sign("finbert", "finbert.sig")
+```
+
+This example generates a signature using a private key based on elliptic curve
+cryptography. It also hashes the model by ignoring `README.md` and any git
+related file present in the model directory.
+
+We also support signing with signing certificates, using a similar API as above.
+
+When verifying, we need to configure the cryptography configuration, so that the
+code knows how to parse the signature.
+
+For the Sigstore example, the simplest verification example would be:
+
+```python
+model_signing.verifying.Config().use_sigstore_verifier(
+    identity=identity, oidc_issuer=oidc_provider
+).verify("finbert", "finbert.sig")
+```
+
+Where `identity` and `oidc_provider` are the parameters obtained after the OIDC
+flow during signing.
+
+To verify the private key example, we could use the following:
+
+```python
+model_signing.verifying.Config().use_elliptic_key_verifier(
+    public_key="key.pub"
+).set_hashing_config(
+    model_signing.hashing.Config().set_ignored_paths(
+        paths=["README.md"], ignore_git_paths=True
+    )
+).verify("finbert", "finbert.sig")
+```
+
+Alternatively, we also support automatic detection of the hashing configuration
+during the verification process. So, the following should also work:
+
+```python
+model_signing.verifying.Config().use_elliptic_key_verifier(
+    public_key="key.pub"
+).verify("finbert", "finbert.sig")
```

Comment:
we don't record ignore paths though? so this wouldn't be an analog of the example above?

Reply:
Oh, no, that's a bug :( Let's fix that for v1.1, and I'll change the example.

Reply:
though you're talking about the sigstore example. feel free to ignore